apis/filesystem/src/main/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImpl.java

```diff
@@ -885,17 +932,26 @@ private void removeDirectoriesTreeOfBlobKey(String container, String blobKey) {
             logger.debug("Could not look for attributes from %s: %s", directory, e);
          }
 
-         String[] children = directory.list();
-         if (null == children || children.length == 0) {
-            try {
-               delete(directory);
-            } catch (IOException e) {
-               logger.debug("Could not delete %s: %s", directory, e);
-               return;
+
+//         String[] children = directory.list();
+//         if (null == children || children.length == 0) {
+         // Don't need to do a listing on the dir, which could be costly. The iterator should be more performant.
+         try {
+            if (isDirEmpty(directory.getPath())) {
+               try {
+                  delete(directory);
+               } catch (IOException e) {
+                  logger.debug("Could not delete %s: %s", directory, e);
```

Comment:
Should this throw an `IOException` instead, possibly wrapped in a `RuntimeException` if necessary?  Repeated below.

Reply:
I believe that the s3 type blob store basically just does nothing when the delete doesn't happen, I'm guessing thats there to be compliant with that?

Reply:
Sorry I missed that this is code movement.  I do think a more narrow exception handling for just removing non-existent files is better but outside the scope of this pull request.